### PR TITLE
Add Location member to SummaryNode instances

### PR DESCRIPTION
**Pull request by joakim-hove**
To enable better error messages

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
I'll merge monday afternoon - unless ...

File: src/opm/output/eclipse/Summary.cpp [hunk 1]
```diff
@@ -1790,13 +1791,15 @@ namespace Evaluator {
...
+void reportUnsupportedKeywords(std::vector<Opm::SummaryNode> keywords)
 {
     std::sort(keywords.begin(), keywords.end());
     auto uend = std::unique(keywords.begin(), keywords.end());
 
-    for (auto kw = keywords.begin(); kw != uend; ++kw)
-        ::Opm::OpmLog::info("Summary keyword '" + *kw + "' is unhandled");
+    for (auto node = keywords.begin(); node != uend; ++node) {
+        const auto& location = node->location();
+        ::Opm::OpmLog::warning("Unhandeled summary keyword '" + node->keyword() + "' at " + location.filename + ", line " + std::to_string(location.lineno));
```
Comment (atgeirr) [inline src/opm/output/eclipse/Summary.cpp]: What if the location was defaulted? Should we not then have a message without the location info?

Also a typo: "Unhandeled" -> "Unhandled"

Comment (joakim-hove) [reply src/opm/output/eclipse/Summary.cpp]: OK - I have added a more explicit content for the auto created locations:
```
Lcoation rst_location("<Added automatically to support restart>", 0);
```

Have fixed the typo.

Comment (atgeirr) [reply src/opm/output/eclipse/Summary.cpp]: Sounds good!

**Comment by atgeirr** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
> I'm going to disagree with @atgeirr here

I agree - and have reverted again. Now I will merge when this is green!

**Comment by atgeirr** [comment]
> I agree - and have reverted again. Now I will merge when this is green!

That is the thing about multiple reviewers... you can choose your poison! I do see his point though and will not complain.

**Pull request merged**